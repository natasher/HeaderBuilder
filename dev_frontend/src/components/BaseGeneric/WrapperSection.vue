<template>
  <div>
    <span class="title">{{ title }}</span>

    <!-- @slot Use this slot for section content -->
    <slot />

  </div>
</template>

<script>
export default {

  name: 'WrapperSection',

  props: {
    /**
     * Title of element to be shown.
     */
    title: {
      type    : String,
      required: true,
    },
  },

}
</script>

