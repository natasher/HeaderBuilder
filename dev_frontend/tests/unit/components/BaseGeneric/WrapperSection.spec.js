import { shallowMount } from '@vue/test-utils'
import WrapperSection from '../../../../src/components/BaseGeneric/WrapperSection'

describe('WrapperSection.vue', () => {

  test('should match snapshot', () => {
    const wrapper = shallowMount(WrapperSection, {
      propsData: {
        title: 'Section title'
      }
    })

    expect( wrapper.element ).toMatchSnapshot()
  })

  test('should render title passed in the props', () => {
    const title   = 'Section title'
    const wrapper = shallowMount(WrapperSection, {
      propsData: {
        title,
      }
    })

    expect( wrapper.find('.title').text() ).toBe( title )
  })

  test('should accept content for default slot', () => {
    const slotContent = '<div id="slot-content">My Awesome Content</div>'
    const wrapper = shallowMount(WrapperSection, {
      propsData: {
        title: 'Section title'
      },
      slots: {
        default: slotContent
      }
    })

    expect( wrapper.contains( '#slot-content' )).toBe( true )
  })

})