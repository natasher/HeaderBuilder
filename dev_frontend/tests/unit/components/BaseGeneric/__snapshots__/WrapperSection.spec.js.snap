// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WrapperSection.vue should match snapshot 1`] = `
<div>
  <span
    class="title"
  >
    Section title
  </span>
   
</div>
`;
